src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs

```diff
@@ -5,33 +5,52 @@ namespace HealthChecks.AzureServiceBus
 {
     public class AzureServiceBusQueueHealthCheck : AzureServiceBusHealthCheck, IHealthCheck
     {
+        private readonly AzureServiceBusOptions _options;
         private readonly string _queueName;
         private string? _connectionKey;
 
-        public AzureServiceBusQueueHealthCheck(string connectionString, string queueName) : base(connectionString)
+        public AzureServiceBusQueueHealthCheck(string connectionString, string queueName, AzureServiceBusOptions options)

```

Comment:
I suggest to use single ctor with options object. You may add all needed arguments as properties into `AzureServiceBusOptions`. Actually I think you will end up with 3 options classes for each health check with  `AzureServiceBusOptions` as base class.

Reply:
Are you thinking of refactoring the AzureServiceBusHealthCheckBuilderExtensions method signatures as well? Or just the AzureServiceBus* ctors?

If you intend moving all parameters of all ctors (connectionstring, endpoint, tokenCredentials, etc) in to one ctor with one options parameter (per AzureServiceBus* class) then I think it will be harder to understand and easier to use it in a "wrong" way.

Can you elaborate a bit more what your thoughts are?

Reply:
I see common design as such - each BlaBlaHC can have single ctor with BlaBlaHCOptions. It minimizes further breaking changes (if any) in public API. Also for each BlaBlaHC you add BlaBlaHCExtensions with at least 1 method accepting BlaBlaHCOptions. Of course you can provide additional methods with any combination of "scalar" arguments for HC. Eventually that arguments will be set as some properties on options object. See `KafkaHealthCheck`.

In case of Azure stuff I see multiple HCs, so of course each of these HCs **may** require different/"personal" options.

> it will be harder to understand and easier to use it in a "wrong" way.

Why? Are there mutually exclusive properties that cannot be set both? If so as I said above - just provide needed extension methods with right combinations that will guide developer. In HC ctor (or even in Execute method if you want, but I would not) you may add additional validation of options object.

Reply:
I must also notice that not all checks are now following this design.

Reply:
Finally back from vacations. I have pushed the options changes. Let's see if this is what you had in mind @sungam3r.